libtransmission/handshake.c

```diff
@@ -642,7 +642,7 @@ readHandshake (tr_handshake    * handshake,
   assert (pstrlen == 19);
   tr_peerIoReadBytes (handshake->io, inbuf, pstr, pstrlen);
   pstr[pstrlen] = '\0';
-  if (memcmp (pstr, "BitTorrent protocol", 19) != 0)
+  if (strncmp (pstr, "BitTorrent protocol", 19) != 0)
```

Comment:
Same note as for changes in clients.c: `pstr` is not pointing to `char` so the warning is issued.